tokenTap/views.py

```diff
@@ -2,6 +2,7 @@
 from django.db import transaction
 from django.http import Http404
 from django.shortcuts import get_object_or_404
+from core.utils import RequestContextExtractor
```

Comment:
You are creating a duplicate instance by instantiating RequestContextExtractor within HasVerifiedCloudflareCaptcha

Reply:
im not passing the whole request object to `HasVerifiedCloudflareCaptcha`

Reply:
im only importing it inside `HasVerifiedCloudflareCaptcha` to declare type safety

Reply:
is not it better to pass request to ConstraintVerification and instantiate it RequestContextExtractor in where you need you to use it?

Reply:
i use it inside `views.py` as the sourcery bot recommended its has a security issue to pass the whole request object to `HasVerifiedCloudflareCaptcha`

Reply:
https://github.com/UnitapApp/unitap-backend/pull/605#discussion_r1730297988

Reply:
ignore it, since we just use extractor in one constraint it is not make sense to instantiate it for all constraints